bundles/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/data/SeverityComparator.java

```diff
@@ -30,7 +28,12 @@ public int compare(DwdWarningData o1, DwdWarningData o2) {
 
         int result = Integer.compare(o1.getSeverity().getOrder(), o2.getSeverity().getOrder());
         if (result == 0) {
-            result = ObjectUtils.compare(o1.getOnset(), o2.getOnset());
+            if (o1.getOnset() == null) {
+                return -1;
+            } else if (o2.getOnset() == null) {
+                return 1;
+            }
```

Comment:
If they are both null then you should return 0.

Reply:
We should return 0 if the reference comparison between both is equal ... :wink: